tuf/repository_lib.py

```diff
@@ -1520,14 +1573,22 @@ def generate_timestamp_metadata(snapshot_filename, version, expiration_date,
   tuf.formats.METADATAVERSION_SCHEMA.check_match(version)
   securesystemslib.formats.ISO8601_DATETIME_SCHEMA.check_match(expiration_date)
   securesystemslib.formats.NAME_SCHEMA.check_match(repository_name)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_length)
+  securesystemslib.formats.BOOLEAN_SCHEMA.check_match(use_hashes)
 
-  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_fileinfo = {}
+
   length, hashes = securesystemslib.util.get_file_details(snapshot_filename,
-      tuf.settings.FILE_HASH_ALGORITHMS)
+    tuf.settings.FILE_HASH_ALGORITHMS)
+
+  length = (use_length and length) or None
+  hashes = (use_hashes and hashes) or None
+
+  # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_version = get_metadata_versioninfo('snapshot', repository_name)
   snapshot_fileinfo[SNAPSHOT_FILENAME] = \
-    tuf.formats.make_fileinfo(length, hashes, version=snapshot_version['version'])
+    tuf.formats.make_metadata_fileinfo(snapshot_version['version'],
+      length, hashes)
```

Comment:
Coding style is to double-indent continuation lines, please fix here and throughout.